bank/persistence/account_operations_test.go

```diff
@@ -0,0 +1,110 @@
+package persistence_test
+
+import (
+	"github.com/iliyanmotovski/bankv1/bank/domain"
+	"github.com/iliyanmotovski/bankv1/bank/persistence"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+	"testing"
+	"time"
+)
+
+func TestItWritesAndReadsAccountDataFromDB(t *testing.T) {
+	session, _ := mgo.DialWithTimeout("localhost:27016", time.Second*10)
+	defer session.Close()
+	session.SetMode(mgo.Monotonic, true)
+
+	accountStore := persistence.NewAccountStore(*session, "bankTest")
+
+	testAccount := domain.Account{UserID: "1234", Currency: "testC", Amount: 10, Type: "VISA"}
+
+	accountID, err := accountStore.InsertAccount(testAccount.UserID, testAccount)
+	if err != nil {
+		t.Error(err.Error())
+	}
+	testAccount.AccountID = accountID
+
+	a, err := accountStore.GetAccounts(testAccount.UserID)
+	if err != nil {
+		t.Error(err.Error())
+	}
+	account := *a
+
+	if account[0] != testAccount {
+		t.Errorf("expected account: %v got: %v", testAccount, account[0])
+	}
+
+	session.DB("bankTest").C("accounts").RemoveAll(nil)
+}
+
+func TestItUpdatesAndDeletesAccountDataFromDB(t *testing.T) {
+	session, _ := mgo.DialWithTimeout("localhost:27016", time.Second*10)
+	defer session.Close()
+	session.SetMode(mgo.Monotonic, true)
+
+	accountStore := persistence.NewAccountStore(*session, "bankTest")
+
+	session.DB("bankTest").C("accounts").Insert(&domain.Account{UserID: "1234", AccountID: "4321", Currency: "testC", Amount: 10})
+
+	testAccountDeposit := domain.Account{UserID: "1234", Currency: "testC", Amount: 5}
+	testAccountWithdraw := domain.Account{UserID: "1234", Currency: "testC", Amount: 2}
+	expected := domain.Account{UserID: "1234", AccountID: "4321", Currency: "testC", Amount: 13}
+
+	err := accountStore.Deposit(testAccountDeposit)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	err = accountStore.Withdraw(testAccountWithdraw)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	account := []domain.Account{}
+	session.DB("bankTest").C("accounts").Find(bson.M{"userid": "1234", "currency": "testC"}).All(&account)
+
+	if account[0] != expected {
+		t.Errorf("expected account: %v got: %v", expected, account[0])
+	}
+
+	err = accountStore.DeleteAccount("1234", "4321")
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	session.DB("bankTest").C("accounts").Find(bson.M{"userid": "1234", "currency": "testC"}).All(&account)
+
+	if len(account) != 0 {
+		t.Error("expected to have 0 accounts in the DB, but there was more")
+	}
+
+	session.DB("bankTest").C("accounts").RemoveAll(nil)
+}
+
+func TestItReadsCorrectHistoryDataFromDB(t *testing.T) {
```

Comment:
It's in the persistence package, so DataFromDB part is not required in the name. 

TestReadsHistoryOfTransactions
TestReadsHistoryOfOneTransaction
TestReadsHistoryOfManyTransactions